src/internal/packager/helm/prepare-render.go

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package helm contains operations for working with helm charts.
+package helm
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/defenseunicorns/zarf/src/internal/packager/template"
+	"github.com/defenseunicorns/zarf/src/pkg/utils"
+	"github.com/defenseunicorns/zarf/src/types"
+	"helm.sh/helm/v3/pkg/releaseutil"
+)
+
+type variableRenderer struct {
+	*Helm
+	values template.Values
+}
+
+func (h *Helm) newVariableRenderer() (*variableRenderer, error) {
+	values, err := template.Generate(h.cfg)
+	if err != nil {
+		return nil, err
+	}
+	values.SetState(&types.ZarfState{})
+
+	return &variableRenderer{
+		Helm:   h,
+		values: *values,
+	}, nil
+}
+
+func (r *variableRenderer) Run(renderedManifests *bytes.Buffer) (*bytes.Buffer, error) {
```

Comment:
This is a decent amount of copy-pasta - would likely be better to split the normal renderer around line 169 to check for a config boolean as to whether it should interact with the cluster.

Reply:
The cluster logic can be pulled into a helper function and then wrapped in an if relatively easily without the logic needing to be here.

Reply:
Separated into two functions as I believe that reads better